sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/ManagedServiceIdentity.Serialization.cs

```diff
@@ -45,7 +50,7 @@ internal void Write(Utf8JsonWriter writer, ModelReaderWriterOptions options, Jso
                 foreach (var item in UserAssignedIdentities)
                 {
                     writer.WritePropertyName(item.Key);
-                    JsonSerializer.Serialize(writer, item.Value);
+                    JsonSerializer.Serialize(writer, item.Value, jsonContext.UserAssignedIdentity);
```

Comment:
Lets swap the serialize calls to a cast to IJsonModel and call the write method on there passing in the MRWOptions and MRWContext